packages/main/src/plugin/container-registry.ts

```diff
@@ -1278,37 +1278,22 @@ export class ContainerProviderRegistry {
   async updateImage(engineId: string, imageId: string, tag: string): Promise<void> {
     let telemetryOptions = {};
     try {
-      // get image info to validate the image can be updated
-      const imageInfo = await this.getMatchingImage(engineId, imageId).inspect();
-
-      // validate the image has tags
-      const repoTags = imageInfo.RepoTags;
-      if (!repoTags?.length) {
-        throw new Error('Image has no tags and cannot be updated');
-      }
-
-      // validate the provided tag exists on this image
-      if (!repoTags.includes(tag)) {
-        throw new Error(`Tag '${tag}' not found on this image`);
+      // return early if the image is a localhost image
+      if (tag.startsWith('localhost/') || tag.startsWith('localhost:')) {
+        throw new LatestImageError('Localhost image is already the latest version');
```

Comment:
I'm not comfortable having the info that the image is not updatable through an error.

I would find clearer to have the function return some structure, with the info of the update:

```
export interface UpdateImageResult {
  updated: boolean;
  oldSha: string;
  newSha: string;
  ...
}

async updateImage(engineId: string, imageId: string, tag: string): Promise<UpdateImageResult>
```